### PR TITLE
Ncgames 4 complete

### PR DESCRIPTION
**Pull request by Wiggy93**
Worked on branch ncgames-4, getReviews and fetchReviews functions are new since ncgames-3 branch along with the test starting on line 26.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+describe('GET commands', () => {
+     test('200 status: GET /api/categories should return an array of categories ', () => {
+        return request(app)
+        .get("/api/categories")
+        .expect(200)
+        .then((response) => {
+            expect(Array.isArray(response.body.Categories)).toBe(true);
+            expect(response.body.Categories[0]).toHaveProperty("slug");
+            expect(response.body.Categories[0]).toHaveProperty("description")
```
Comment (MKD01) [inline __tests__/app.test.js]: These assertions are simply checking only the first category, i would recommend iterating through the categories array and checking every category is what we expect.

File: app/app.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const express = require("express");
+const app = express();
+const { getCategory, getReviews} = require("../controllers/app.controller")
+
+app.use(express.json())
```
Comment (MKD01) [inline app/app.js]: This is not needed yet so it should not be added right now

File: models/app.model.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  
+    const reviewCommentCount = 
+    `SELECT reviews.*, COUNT (comments.comment_id) AS comment_count  
+    FROM reviews 
+    LEFT JOIN comments ON comments.review_id=reviews.review_id 
+    GROUP BY reviews.review_id
+    ORDER BY created_at DESC;`;
+    
+    return db.query(reviewCommentCount).then((result) => {
+        console.log(result.rows)
```
Comment (MKD01) [inline models/app.model.js]: Please remove console.log

**Review by MKD01 — APPROVED** [review]
Happy for this to be merged after those changes have been made.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+                expect(review).toHaveProperty("review_id");
+                expect(review).toHaveProperty("category");
+                expect(review).toHaveProperty("review_img_url");
+                expect(review).toHaveProperty("created_at");
+                expect(review).toHaveProperty("votes");
+                expect(review).toHaveProperty("designer");
+                expect(review).toHaveProperty("comment_count");
+            })
+            
+            expect(reviews).toBeSortedBy("created_at", {descending : true,});
```
Comment (mezzDavies) [inline __tests__/app.test.js:54]: 👍 😄

**Review by mezzDavies — APPROVED** [review]
Yep please feel free to merge 👍

**Pull request merged**